### PR TITLE
Add optional Owner to event subscriptions

### PR DESCRIPTION
**Pull request by Francescu**
Hey 👋

Inspired by [Observable-Swift](https://github.com/slazyk/Observable-Swift), I added the support of optionally add an `owner` to the subscription. It allows to easily automatically unsubscribe based on `owner` life cycle. 

There's no API break for existing Interstellar users.

### CODE REVIEW

**Comment by JensRavens** [comment]
Unsubscribing seems to be quite a long running topic (about 90% of all requested changes are about that). While I like the idea of coupling the lifetime of an observable to another object, this would push the library from "this is just swift using ARC" to "it's magic, trust us" for one simple reason: One of my favorite parts of Swift is that memory management is predictable. You nil out a variable and the value get's deallocated. With this the handler gets deallocated the next time the observable gets updated which might be minutes later - so instead of crashing immediately due to some bug it gets triggered in a later iteration, making it really hard to debug.

To sum it up: I think this functionality is exceeding the basic idea of "this is as simple as possible" that this library tries to achieve. If the library would follow this path, it would end up being RxSwift. So let me think about it for a few more days if the same result could be achieved with another way.
